patches/tModLoader/Terraria/ModLoader/ItemLoader.cs

```diff
@@ -1431,89 +1431,108 @@ public static bool CanStackInWorld(Item increase, Item decrease)
+	/// <param name="destination">The item instance that <paramref name="source"/> will attempt to stack onto</param>
+	/// <param name="source">The item instance being stacked onto <paramref name="destination"/></param>
+	/// <param name="numTransferred">The quanity of <paramref name="source"/> that was transferred to <paramref name="destination"/></param>
+	/// <param name="infiniteSource">If true, <paramref name="source"/>.stack will not be decreased</param>
+	/// <param name="numToTransfer">
+	/// An optional argument used to specify the quantity of items to transfer from <paramref name="source"/> to <paramref name="destination"/>.<br/>
+	/// By default, as many items as possible will be transferred.  That is, either source will be empty, or destination will be full (maxStack)
+	/// </param>
+	public static void StackItems(Item destination, Item source, out int numTransferred, bool infiniteSource = false, int? numToTransfer = null)
 	{
-		numTransfered = numToTransfer ?? Math.Min(decrease.stack, increase.maxStack - increase.stack);
+		numTransferred = numToTransfer ?? Math.Min(source.stack, destination.maxStack - destination.stack);
 
-		foreach (var g in HookOnStack.Enumerate(increase.globalItems)) {
-			g.OnStack(increase, decrease, numTransfered);
+		OnStack(destination, source, numTransferred);
+
+		if (source.favorited) {
+			destination.favorited = true;
+			source.favorited = false;
 		}
 
-		increase.ModItem?.OnStack(decrease, numTransfered);
+		destination.stack += numTransferred;
+		if (!infiniteSource)
+			source.stack -= numTransferred;
+	}
 
-		if (decrease.favorited) {
-			increase.favorited = true;
-			decrease.favorited = false;
+	/// <summary>
+	/// Calls the GlobalItem.OnStack hooks in <paramref name="destination"/>, then the ModItem.OnStack hook in <paramref name="destination"/>
+	/// </summary>
+	/// <param name="destination">The item instance that <paramref name="source"/> will attempt to stack onto</param>
+	/// <param name="source">The item instance being stacked onto <paramref name="destination"/></param>
+	/// <param name="numToTransfer">The quanity of <paramref name="source"/> that will be transferred to <paramref name="destination"/></param>
+	public static void OnStack(Item destination, Item source, int numToTransfer)
+	{
+		foreach (var g in HookOnStack.Enumerate(destination.globalItems)) {
+			g.OnStack(destination, source, numToTransfer);
 		}
 
-		increase.stack += numTransfered;
-		if (!infiniteSource)
-			decrease.stack -= numTransfered;
+		destination.ModItem?.OnStack(source, numToTransfer);
 	}
 
 	private static HookList HookSplitStack = AddHook<Action<Item, Item, int>>(g => g.SplitStack);
 
-	public static Item TransferWithLimit(Item decrease, int limit)
+	/// <summary>
+	/// Creates a clone of <paramref name="source"/>, then transfers at most <paramref name="limit"/> from <paramref name="source"/> to the clone
```

Comment:
Hmm, not confident we should express partial implementation details here. Consider:
```
Extract up to <paramref name="limit"/> items from <paramref name="source"/>. If some items remain, <see cref="SplitStack" /> will be used. 
```

Reply:
good idea